File: 1000 and 1 tasks/data-structure1/ransom-note.cs
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {
        var table = new int[26];

        foreach(var c in ransomNote) {
            table[c - 'a']--;
        }
        foreach(var c in magazine) {
            table[c - 'a']++;
        }

        return table.All(x => x >= 0);
    }
}

File: 1000 and 1 tasks/new-top-150/ransom-note.java
import java.util.Arrays;

public class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        var table = new int[26];
        for (var ch : magazine.toCharArray()) {
            table[ch - 'a']++;
        }
        for (var ch : ransomNote.toCharArray()) {
            table[ch - 'a']--;
        }

        return Arrays.stream(table).allMatch(count -> count >= 0);
    }
}